java-checks/src/main/java/org/sonar/java/checks/StringPrimitiveConstructorCheck.java

```diff
@@ -53,19 +60,83 @@ protected MethodMatchers getMethodInvocationMatchers() {
       MethodMatchers.create().ofTypes("java.lang.Float").constructor().addParametersMatcher("float").build(),
       MethodMatchers.create().ofTypes("java.lang.Double").constructor().addParametersMatcher("double").build(),
       MethodMatchers.create().ofTypes("java.lang.Boolean").constructor().addParametersMatcher("boolean").build(),
-      MethodMatchers.create().ofTypes("java.math.BigInteger").constructor().addParametersMatcher(STRING).build());
+      MethodMatchers.create().ofTypes(BIG_INTEGER).constructor().addParametersMatcher(STRING).build());
   }
 
   @Override
   protected void onConstructorFound(NewClassTree newClassTree) {
+    if (newClassTree.classBody() != null) {
+      return;
+    }
     if(isBigIntegerPotentiallyBiggerThanLong(newClassTree)) {
       return;
     }
-    reportIssue(newClassTree.identifier(), "Remove this \"" + newClassTree.symbolType().name() + "\" constructor");
+    ((InternalJavaIssueBuilder) ((DefaultJavaFileScannerContext) context).newIssue())
+      .forRule(this)
+      .onTree(newClassTree.identifier())
+      .withMessage("Remove this \"" + newClassTree.symbolType().name() + "\" constructor")
```

Comment:
You can use the parametrized `withMessage()` method. It embeds String.format() already:
```java
.withMessage("Remove this \"%s\" constructor", newClassTree.symbolType().name())
```
I feel it makes the issue message easier to read